### PR TITLE
shell-completion: add bash completion for userdbctl

### LINKED ISSUE #35660: Missing bash completions for `userdbctl` arguments
### systemd version the issue has been seen with

257-1

### Used distribution

Arch Linux

### Linux kernel version used

6.12.4-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

userdbctl

### Expected behaviour you didn't see

Double-tapping TAB after typing `userdbctl ` returns auto-completions for `user`, `group`, etc.

### Unexpected behaviour you saw

Double-tapping TAB after typing `userdbctl ` returns a list of the files in the current working directory.

### Steps to reproduce the problem

1. Type `userdbctl ` while in a bash shell.
2. Hit "TAB" twice and see a list of the files in the current working directory, instead of `userdbctl`'s arguments.

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### PR DESCRIPTION
**Pull request by jdholtz**
Closes #35660.

### CODE REVIEW

**Comment by bluca** [comment]
Needs to be listed in `shell-completion/bash/meson.build` otherwise it won't be installed

**Review by bluca — APPROVED** [review]

File: shell-completion/bash/userdbctl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+# along with systemd; If not, see <https://www.gnu.org/licenses/>.
+
+__contains_word () {
+    local w word=$1; shift
+    for w in "$@"; do
+        [[ $w = "$word" ]] && return
+    done
+}
+
+__get_all_users () { userdbctl user --no-legend -B | { while read -r a b; do printf "%s\n" "$a"; done; } ; }
```
Comment (yuwata) [inline shell-completion/bash/userdbctl]: Please break the line. Also, use `local a b` for safety.

File: shell-completion/bash/userdbctl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+__get_all_groups () { userdbctl group --no-legend -B | { while read -r a b; do printf "%s\n" "$a"; done; } ; }
```
Comment (yuwata) [inline shell-completion/bash/userdbctl]: Also here. Moreover, please insert an empty line between function defs.

File: shell-completion/bash/userdbctl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    if __contains_word "$prev" ${OPTS[ARG]}; then
+        case $prev in
+            --output)
+                comps=$( userdbctl --output=help 2>/dev/null )
+                ;;
+            -s|--service|--uid-min|--uid-max)
+                comps=''
+                ;;
+            --with-nss|--synthesize|--with-dropin|--with-varlink|--multiplexer|--boundaries)
+                comps='true false'
```
Comment (yuwata) [inline shell-completion/bash/userdbctl]: Please use yes and no.

File: shell-completion/bash/userdbctl [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        comps=$( __get_all_groups )
+    elif __contains_word "$verb" ${VERBS[STANDALONE]}; then
+        comps=''
+    fi
+
+    COMPREPLY=( $(compgen -W '$comps' -- "$cur") )
+    return 0
+}
+
+
```
Comment (yuwata) [inline shell-completion/bash/userdbctl]: Doubled empty lines

**Review by yuwata — CHANGES_REQUESTED** [review]

**Pull request merged**